### DISCUSSION TITLE
gltfpack impacting .glb MetallicRoughnessOcclusion texture

### DISCUSSION THREAD

**Discussion by Bketting** [zeux/meshoptimizer#251] [Q&A] [answered]
Original .glb 3D model has a color, normal, and MRO texture, with the last one structured as G=roughness, B=metallic, R=occlusion. Uploading the file in the model-viewer editor results in the correct formatting. See the picture below:

_Image removed_

Running gtlfpack on the .glb with simpe -i .glb -o .glb command line results in .glb that is highly glossy in the model-viewer editor. No other changes were made to the .glb.

_Image removed_

Any ideas on how to fix this? Turning of the normal map in model-viewer editor solves the problem, but also creates a new one (no normals)

**Comment by zeux** [comment]
That seems like a bug. Would you mind opening an issue with a source model file attached if possible? If you aren't able to share the model publicly you can email it to arseny.kapoulkine@gmail.com

**Comment by zeux** [comment, marked as answer]
After an offline investigation this was determined to be a bug in three.js that had a rather unique set of conditions so it only manifested in model-viewer. The fix for the bug has been submitted to three.js repository: https://github.com/mrdoob/three.js/pull/21511
